Utils.ps1

```diff
@@ -22,6 +22,9 @@ Set-Alias ?? Invoke-NullCoalescing -Force
 function Invoke-Utf8ConsoleCommand([ScriptBlock]$cmd) {
     $currentEncoding = [Console]::OutputEncoding
     try {
+        # A native executable that writes to stderr AND has its stderr redirected will generate non-terminating
+        # error records if the user has set $ErrorActionPreference to Stop. Override that value in this scope.
+        $ErrorActionPreference = 'Continue'

```

Comment:
Asking my fellow PowerShell MVPs there is no good way around this.  We can clean our errors out of $Error and we should probably stash them in a private module variable ($poshErrors?).  There are ways to extract the value of private module variables IIRC.  This might help in certain debugging scenarios.

Regarding the rewrite above A) I'm heading out to Disneyland tomorrow with the kids so I won't have a chance to look at this until Sunday night and B) I think we can avoid calling `git rev-parse` if we find `.git` right?  That is essentially what the current approach does.  OTOH if we find `HEAD` and `refs` dir, then we call `git rev-parse`.   For the handling of $Error, we could put that in either the Invoke-Utf8ConsoleCommand or the prompt function?  Thoughts?